js/geo/transform.js

```diff
@@ -375,7 +394,8 @@ Transform.prototype = {
                 y2 !== undefined ? y2 : this.y));
         }
 
-        this._constraining = false;
+        this._unmodified = this._constraining.unmodified;
+        this._constraining = undefined;
```

Comment:
OK, I pushed https://github.com/mapbox/mapbox-gl-js/commit/7ffbb40ed598af23868e63ad0db19e22da9fe153 to show you what I mean. It passes tests. Feel free to merge into your branch if it makes sense.

`Map#resize` no longer calls `_constrain`, it's now implicitly called by `transform.resize` after my recent refactoring.

It might be worth refactoring `Transform` later, but that's a subject for a separate discussion — feel free to open an issue about this.